### PR TITLE
Remove dependency of LanguageServerProtocol on SKSupport

### PR DESCRIPTION
**Pull request by ahoppen**
Apart from removing unnecessary imports, the following changes were
performed:
- Move `CancellationToken` from `SKSupport` to `LanguageServerProtocol`
- Move `CustomCodable` to its own module
  - `CustomCodable` belongs in LanguageServerProtocol but moving it there causes compilation to fail (rdar://57088846). As a workaround, move it to its own module.

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -85,6 +85,11 @@ let package = Package(
       .testTarget(
         name: "SKSupportTests",
         dependencies: ["SKSupport", "SKTestSupport"]),
+
+      // Workaround for rdar://57088846. If CustomCodable lives in the LanguageServerProtocol the code fails to compile. Move CustomCodable to LanguageServerProtocol and delete this module once the problem is fixed.
+      .target(
+        name: "CustomCodableModule"
```
Comment (benlangmuir) [inline Package.swift]: Do we get ambiguity if we just call this `CustomCodable`?

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test macOS platform

**Pull request merged**